mscore/workspace.cpp

```diff
@@ -220,12 +220,11 @@ void MuseScore::changeWorkspace(const QString& name)
 
 void MuseScore::changeWorkspace(Workspace* p, bool first)
       {
-      if (!first) {
+      if (!first)
             WorkspacesManager::currentWorkspace()->save();
-            if (WorkspacesManager::currentWorkspace())
-                  disconnect(getPaletteWorkspace(), &PaletteWorkspace::userPaletteChanged, WorkspacesManager::currentWorkspace(), QOverload<>::of(&Workspace::setDirty));
-            }
 
+      if (WorkspacesManager::currentWorkspace())
+            disconnect(getPaletteWorkspace(), &PaletteWorkspace::userPaletteChanged, WorkspacesManager::currentWorkspace(), QOverload<>::of(&Workspace::setDirty));
```

Comment:
Yes, this was the other approach I tried and it seems to work as well.  I have a question about the order of the default workspaces, it seems we sometimes assume is always basic then advanced but reality can be otherwise, maybe this changes helps ensure basic does come first?  Still, I've seen other cases - not just when advanced is selected in startup - where it shows first in combo box.